src/components/Tooltip/Tooltip.test.tsx

```diff
@@ -0,0 +1,30 @@
+import "@testing-library/jest-dom/extend-expect"
+import React from "react"
+import { render } from "@testing-library/react"
+
+import Tooltip from "./Tooltip"
+
+describe("Tooltip", () => {
+  it("renders successfully", () => {
+    const { getByTestId } = render(
+      <Tooltip title="Title">
+        <button>hover me</button>
+      </Tooltip>
+    )
+
+    /**
+     * Check if the element exists
+     */
+    expect(getByTestId("tooltip")).toBeInTheDocument()
+  })
+
+  it("renders the children succesfully", () => {
+    const { getByText } = render(
+      <Tooltip title="Title">
+        <button>hover me</button>
+      </Tooltip>
+    )
+
+    expect(getByText("hover me")).toBeInTheDocument()
+  })
+})
```

Comment:
Any way we can add a test that checks if it's not visible before hovering and it becomes visible after hovering?
I'm not sure how easy it is to simulate hover with testing-library

Reply:
Just saw that there is "fireEvent.mouseOver(element)", which should work for this

Reply:
Then we're gonna be testing styles. Maybe we can use something like jest-styled-components

Reply:
I mean the hovering thing is being done through CSS

Reply:
https://stackoverflow.com/questions/52377301/test-styles-triggered-by-mouse-over-events

Reply:
> fireEvent.mouseOver is not actually performing the changes in the dom that you're expecting it to.

https://github.com/testing-library/jest-dom/issues/59

Anyways I'll check more into this. Maybe some workaround

Reply:
You're right that there's no point in testing the styles, `jest-styled-components` is more for testing if the correct styles were applied, which I don't find as useful, as these kind of tests are basically testing implementation and are very brittle.

The point of my suggestion is to test the functionality of the tooltip showing up if the element is hovered over, which is the basic function of the component, and we should know if that breaks in the future by having a test for it. It shouldn't matter to the test how it's happening (with css, js or anything), just that it happens.

The test would basically be: 
- render the component
- the tooltip shouldn't be visible at the start: `expect(getByText("tooltip text")).not.toBeVisible()`
- then hover over the element `fireEvent.mouseOver`
- and then it should show up: `expect(getByText("tooltip text")).toBeVisible()`

refs: https://github.com/testing-library/jest-dom#tobevisible
https://testing-library.com/docs/guide-disappearance

Reply:
Yep I agree. I'm just documenting, in case I pause and come back to this.

I tried the `toBeVisible` that you suggested, and it's not working. The problem is something with the mouseOver not changing the styles as expected

Reply:
Some possible solution would be: https://testing-library.com/docs/ecosystem-user-event/#hoverelement

But no success anyway. The hovering effect through CSS is not being able to be tested.

Since this isn't a very huge problem, we're going to take it as an exception.